src/core/enums/Action.enum.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * Xournal++
+ *
+ * Enum for all actions
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <string>
+
+#include <glib.h>  // for g_warning
+
+#include "util/Assert.h"
+#include "util/StringUtils.h"
+
+/******************************************************************************
+*******************************************************************************
+
+RUN THE GENERATOR IF YOU CHANGE THIS FILE!
+
+php generateConvertNEW.php
+
+*******************************************************************************
+******************************************************************************/
+
+enum class Action {
```

Comment:
Should either be `size_t` or `uint32_t`, because you are casting it very often to a `size_t`.
```suggestion
enum class Action : size_t {
```

With that you can use:
```
template< class Enum >
constexpr [std::underlying_type_t](http://en.cppreference.com/w/cpp/types/underlying_type)<Enum> to_underlying( Enum e ) noexcept
```